packages/models/src/actor/interface.ts

```diff
@@ -2,15 +2,24 @@ import { Behavior } from '../utils'
 
 export type ActorName = string | symbol
 export type ActorURI = string
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type ConstructorFunction = new (...args: Array<any>) => unknown
 
+export interface ActorRefParam {
+  param: string
+  index: number
+  value?: string
+}
 export interface ActorRef {
   name: ActorName
   protocol: string
   path: string
   uri: ActorURI
+  params: Map<string, ActorRefParam>
 }
 
-export type MessagePayload<Payload = string | number | object> = {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type MessagePayload<Payload = any> = {
```

Comment:
Why fallback type to `any`

Reply:
> Why fallback type to `any`

Because the error appears when the `find` is called, like followe:

```typescript
const actor = registry.find<Model>(ref)
```

Reply:
Could you paste the error message in this thread

Reply:
> Could you paste the error message in this thread

As I changed the `find` method in `Registry` from `find<T = Actor>()` to `find<T extends Actor = Actor>` for the suggestion https://github.com/ckb-js/kuai/pull/181#discussion_r1131831929, I got this error:

```
 src/app.controller.ts(23,48): error TS2344: Type 'OmnilockModel' does not satisfy the constraint 'Actor<unknown, MessagePayload<string | number | object>>'.
         Types of property 'handleCall' are incompatible.
           Type '(msg: ActorMessage<MessagePayload<StoreMessage>>) => void' is not assignable to type '(_msg: ActorMessage<MessagePayload<string | number | object>>) => void'.
             Types of parameters 'msg' and '_msg' are incompatible.
               Type 'ActorMessage<MessagePayload<string | number | object>>' is not assignable to type 'ActorMessage<MessagePayload<StoreMessage>>'.
                 Type 'MessagePayload<string | number | object>' is not assignable to type 'MessagePayload<StoreMessage>'.
                   Type '{ pattern: string; value?: string | number | object | undefined; }' is not assignable to type 'MessagePayload<StoreMessage>'.
                     Types of property 'value' are incompatible.
                       Type 'string | number | object | undefined' is not assignable to type 'StoreMessage | undefined'.
                         Type 'string' is not assignable to type 'StoreMessage | undefined'.
       src/app.controller.ts(37,48): error TS2344: Type 'OmnilockModel' does not satisfy the constraint 'Actor<unknown, MessagePayload<string | number | object>>'.
       src/app.controller.ts(51,46): error TS2344: Type 'RecordModel' does not satisfy the constraint 'Actor<unknown, MessagePayload<string | number | object>>'.
         Types of property 'handleCall' are incompatible.
           Type '(msg: ActorMessage<MessagePayload<StoreMessage>>) => void' is not assignable to type '(_msg: ActorMessage<MessagePayload<string | number | object>>) => void'.
       src/app.controller.ts(64,46): error TS2344: Type 'RecordModel' does not satisfy the constraint 'Actor<unknown, MessagePayload<string | number | object>>'.
       src/app.controller.ts(75,46): error TS2344: Type 'RecordModel' does not satisfy the constraint 'Actor<unknown, MessagePayload<string | number | object>>'.
       src/app.controller.ts(83,46): error TS2344: Type 'RecordModel' does not satisfy the constraint 'Actor<unknown, MessagePayload<string | number | object>>'.
```

So I made this modification

Reply:
```
Type 'string | number | object | undefined' is not assignable to type 'StoreMessage | undefined'.
  Type 'string' is not assignable to type 'StoreMessage | undefined'.
```

This is the key problem, `string` is incompatible with `StoreMessage`, any idea on how to make them compatible @yanguoyu

Reply:
> ```
> Type 'string | number | object | undefined' is not assignable to type 'StoreMessage | undefined'.
>   Type 'string' is not assignable to type 'StoreMessage | undefined'.
> ```
> 
> This is the key problem, `string` is incompatible with `StoreMessage`, any idea on how to make them compatible @yanguoyu

I think maybe we should better not to give the type limit to develop to define the `Payload`

Reply:
> > ```
> > Type 'string | number | object | undefined' is not assignable to type 'StoreMessage | undefined'.
> >   Type 'string' is not assignable to type 'StoreMessage | undefined'.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This is the key problem, `string` is incompatible with `StoreMessage`, any idea on how to make them compatible @yanguoyu
> 
> I think maybe we should better not to give the type limit to develop to define the `Payload`

The message payload will not be interchangeable between different projects without a schema

Reply:
`find = <T extends Actor<unknown, MessagePayload<any>> = Actor>(uri: ActorURI)`
define find's T like this is enough.

Reply:
> `find = <T extends Actor<unknown, MessagePayload<any>> = Actor>(uri: ActorURI)` define find's T like this is enough.

I'll have a try

Reply:
> `find = <T extends Actor<unknown, MessagePayload<any>> = Actor>(uri: ActorURI)` define find's T like this is enough.

@Keith-CY Here it is